### PR TITLE
Initial support for MemTiles

### PR DESCRIPTION
**Pull request by jgmelber**

### CODE REVIEW

File: lib/Dialect/AIE/IR/AIEDialect.cpp [hunk 1]
```diff
@@ -1137,6 +1158,27 @@ int SwitchboxOp::getNumDestConnections(WireBundle bundle) {
     default:
       return 0;
     }
+  else if (getTileOp().isMemTile())
+    switch (bundle) {
+    //case WireBundle::Core:
```
Comment (fifield) [inline lib/Dialect/AIE/IR/AIEDialect.cpp:1163]: What is the purpose of the commented lines here and in getNumSourceConnections ?

**Review by fifield — APPROVED** [review]

File: lib/Dialect/AIE/IR/AIEDialect.cpp (hunk 1 above)
Comment (jgmelber) [reply lib/Dialect/AIE/IR/AIEDialect.cpp]: Matching the style of prior code. Will remove.

File: lib/Dialect/AIE/IR/AIEDialect.cpp [hunk 2]
```diff
@@ -1096,6 +1093,19 @@ int SwitchboxOp::getNumSourceConnections(WireBundle bundle) {
...
     }
+  else if (getTileOp().isMemTile())
+    switch (bundle) {
+    case WireBundle::DMA:
+      return 6;
+    case WireBundle::North:
+      return 4;
+    case WireBundle::South:
+      return 6;
+    case WireBundle::Trace:
```
Comment (stephenneuendorffer) [inline lib/Dialect/AIE/IR/AIEDialect.cpp:1104]: I thought I had plumbed this through the targetmodel already? Oops, guess I never committed it!

File: lib/Targets/AIETargetXAIEV2.cpp [hunk 3]
```diff
@@ -374,6 +374,189 @@ mlir::LogicalResult AIETranslateToXAIEV2(ModuleOp module, raw_ostream &output) {
...
+      }
+
+      int bdNum = blockMap[&block];
+      if (foundBd) {
+        // TODO AB mode separated
+
+        // TODO For now, we are going to name each dma desc with loc and bd
+        // which we assume is unique. This is strictly not enforced but in
+        // practice, this is true
+        output << "XAie_DmaDesc " << tileDMAInstStr(col, row, bdNum) << ";\n";
```
[the remark this answers is not in the document]
Comment (jgmelber) [reply lib/Targets/AIETargetXAIEV2.cpp:496]: For MemTile or for the highlighted piece of code?

[the remark this answers is not in the document]
Comment (stephenneuendorffer) [reply lib/Targets/AIETargetXAIEV2.cpp:496]: Particularly for this target code... For instance, something like aie/test/Targets/AIEGenerateXAIE/tileDMA.mlir, but for a memtile instead.

**Review by stephenneuendorffer — APPROVED** [review]

**Pull request merged**